gossip/blockproc/drivermodule/driver_txs.go

```diff
@@ -128,14 +128,20 @@ func (p *DriverTxTransactor) PopInternalTxs(_ iblockproc.BlockCtx, _ iblockproc.
 	return internalTxs
 }
 
-func (p *DriverTxListener) OnNewReceipt(tx *types.Transaction, r *types.Receipt, originator idx.ValidatorID) {
+func (p *DriverTxListener) OnNewReceipt(tx *types.Transaction, r *types.Receipt, originator idx.ValidatorID, baseFee *big.Int) {
 	if originator == 0 {
 		return
 	}
 	originatorIdx := p.es.Validators.GetIdx(originator)
 
 	// track originated fee
-	txFee := new(big.Int).Mul(new(big.Int).SetUint64(r.GasUsed), tx.GasPrice())
+	var gasPrice *big.Int
+	if p.es.Rules.Upgrades.Allegro {
+		gasPrice = effectiveGasPrice(tx, baseFee)
+	} else {
+		gasPrice = tx.GasPrice()
+	}
+	txFee := new(big.Int).Mul(new(big.Int).SetUint64(r.GasUsed), gasPrice)
```

Comment:
Although we do not support it right now, the blob gas price should be considered here as well.